net/ollama/Portfile

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        ollama ollama 0.3.6 v
+revision            0
+
+categories          net
+maintainers         @rdallman \
+                    openmaintainer
+license             MIT
+
+description         Ollama
+long_description    Ollama runs and manages LLMs
+homepage            https://ollama.com
+
+depends_build       port:cmake \
+                    port:go
+fetch.type          git
+
+use_configure       no
+build               {
+    # fix 'ollama --version'
+    reinplace "s/var Version string = \".*\"/var Version string = \"${version}\"/" version/version.go
+
+    system -W ${worksrcpath} "go generate ./..."
+    system -W ${worksrcpath} "go build ."
```

Comment:
the port is downloading packages in the the `build` phase, that's not what should happen - we have a `fetch` phase for that. Perhaps you can use the `golang` PortGroup and specify the packages. I don't have much experience with it, but perhaps @herbygillot can provide some additional guidance here?!

Reply:
Yah, the golang group is probably the right thing.

Reply:
yea good eye thanks. I will play with the golang portgroup, I had no dice the last time due to the issues noted https://github.com/macports/macports-ports/pull/25274#issuecomment-2294316865 about using release zip but should be fine for dependencies.